File: Distancia3PP/Punto2d.vb
Public Class Punto2d
    Inherits Punto
    Protected y As Decimal

    Public Property Y1() As Decimal
        Get
            Return y
        End Get
        Set(value As Decimal)
            y = value
        End Set
    End Property

    Public Sub New()
        MyBase.New()
        y = 0
    End Sub

    Public Sub New(x As Decimal, y As Decimal)
        MyBase.New(x)
        Y1 = y
    End Sub

    Public Overloads Function Calcular_Distancia(a As Punto2d)
        Return Math.Sqrt(Math.Pow(x - a.X1, 2) + Math.Pow(y - a.Y1, 2))
    End Function

    Public Overrides Function ToString() As String
        Return x + " " + y
    End Function
End Class

File: Distancia3P/Punto2d.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Distancia3P
{
    class Punto2d : Punto
    {
        protected decimal y;

        public decimal Y
        {
            get { return y; }
            set { y = value; }
        }

        public Punto2d() : base()
        {
            y = 0;
        }
        public Punto2d(decimal x, decimal y) : base(x)
        {
            this.y = y;
        }

        public decimal Calcular_Distancia(Punto2d a)
        {
            return (decimal)(Math.Sqrt(Math.Pow((double)(x - a.X), 2) + Math.Pow((double)(y - a.Y), 2)));

        }

        public override string ToString()
        {
            return x + " " + y;
        }
    }
}

